fastify.js

```diff
@@ -298,6 +299,11 @@ function build (options) {
       }
     })
   })
+  avvio.ready(function (err, instance, done) {
```

Comment:
Calling ready() would make this start. Avvio has a 'preReady' event which is synchronous.  I think the best path to have this feature is to use that instead.

Reply:
🤔 I'll give that a try this evening unless someone gets to it before me.

Reply:
I think it might require a change in avvio because that is sync (so no callbacks9

Reply:
Oof 😢

Reply:
To archive this in the other PR I have to rewrite the `ready` function managing the hooks' async execution:

https://github.com/fastify/fastify/pull/2166/files#diff-20da0b6140a4768d182af0d9b185ac28R265

I have still one issue with the timeout loading (https://github.com/mcollina/avvio/issues/103), then, I think that that piece of code could be cherry picked here and all should works

I would keep the `globalHooks` usage in `v2`